### PR TITLE
feat : Add SharedPreferences Util

### LINKED ISSUE #1832: Create Util for Shared Preferences to use it from all over the app
**Actual Behaviour**

Currently shared preference is used in different different activities and classes which increases complexity. 

**Expected Behaviour**

Create Util class for shared preference which will be used to save data in shared preferences all over the app. 

**Would you like to work on the issue?**

Yes

**Target Milestone to solve the issue**

[Milestone 11](https://github.com/fossasia/open-event-android/milestone/34)

### PR DESCRIPTION
**Pull request by Shailesh351**
Fixes #1832 

Changes: Add SharedPreferencesUtil to use it from all over the App.

### CODE REVIEW

**Comment by Shailesh351** [comment]
@iamareebjamal @rohanagarwal94 @arp95 @shreyasnbhat @mahikaw Please review

File: android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java [hunk 1]
```diff
@@ -142,7 +138,7 @@ private void showEvent() {
             return;
 
         try {
-            int id = sharedPreferences.getInt(ConstantStrings.SESSION_MAP_ID,-1);
+            int id = SharedPreferencesUtil.getInt(ConstantStrings.SESSION_MAP_ID,-1);
```
Comment (iamareebjamal) [inline android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java]: Space after comma

File: android/app/src/main/java/org/fossasia/openevent/utils/SharedPreferencesUtil.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package org.fossasia.openevent.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import org.fossasia.openevent.OpenEventApp;
+
+public class SharedPreferencesUtil {
+    private static SharedPreferences sharedPreferences;
+    private static SharedPreferences.Editor editor;
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/SharedPreferencesUtil.java:10]: Storing context holders in static variables is a memory leak. Create a static method of `getInstance()` and return an instance of this object

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/utils/SharedPreferencesUtil.java (hunk 2 above)
Comment (Shailesh351) [reply android/app/src/main/java/org/fossasia/openevent/utils/SharedPreferencesUtil.java:10]: @iamareebjamal I'm not storing context here. I am making static SharedPreference object which is returned by context.getSharedPreferences() method. It should not create a memory leak. Right ?

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/utils/SharedPreferencesUtil.java:10]: SharedPreference holds context as it is dependent on it, but here it is application context so the leak does not happen

**Review by iamareebjamal — APPROVED** [review]

**Review by rohanagarwal94 — APPROVED** [review]

**Review by shreyasnbhat — APPROVED** [review]

**Pull request merged**